model/event/subscriptions.go

```diff
@@ -216,74 +227,85 @@ const (
 	SelectorInBuild      = "in-build"
 )
 
+var filterKeyForSelectorType = map[string]string{
+	SelectorObject:       filterObjectKey,
+	SelectorID:           filterIDKey,
+	SelectorProject:      filterProjectKey,
+	SelectorOwner:        filterOwnerKey,
+	SelectorRequester:    filterRequesterKey,
+	SelectorStatus:       filterStatusKey,
+	SelectorDisplayName:  filterDisplayNameKey,
+	SelectorBuildVariant: filterBuildVariantKey,
+	SelectorInVersion:    filterInVersionKey,
+	SelectorInBuild:      filterInBuildKey,
+}
+
 // FindSubscriptions finds all subscriptions of matching resourceType, and whose
-// selectors match the selectors slice
-func FindSubscriptions(resourceType string, selectors []Selector) ([]Subscription, error) {
-	if len(selectors) == 0 {
+// filter and regex selectors match the attributes of the event.
+func FindSubscriptions(resourceType string, eventAttributes map[string][]string) ([]Subscription, error) {
```

Comment:
Should this be called `FindSubscriptionsByAttributes`?

Reply:
np. done